sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGenerateExec.scala

```diff
@@ -826,13 +827,99 @@ class GpuGenerateIterator(
+          // single row, we need to actually add row duplicates, then split
+          val tbl = withResource(spillable.getColumnarBatch()) { src =>
+            GpuColumnVector.from(src)
+          }
+          withResource(tbl) { _ =>
+            if (tbl.getColumn(generatorOffset).getNullCount == 1) {
+              // 1 row, and the element to explode is null, we cannot
+              // handle this case
+              throw new GpuSplitAndRetryOOM(
+                s"GPU OutOfMemory: cannot split a batch of 1 rows when " +
+                  s"the array to explode is null!")
+            }
+            // borrow the column to explode, in order to split it as a table
+            // explodedTbl is 1 row per element of the list
+            val explodedTbl =
+            withResource(new Table(tbl.getColumn(generatorOffset))) { generatorTmpTable =>
+              // we use regular `explode` because we know we have 1 list column
+              // and it isn't null => this operation will produce N rows, 1 row per element
+              // of the column at `generatorOffset`
+              generatorTmpTable.explode(0)
+            }
+
+            // Because we are likely to be here solving a cuDF column length limit
+            // we are just going to split in half blindly until we can fit it.
+            val splits = withResource(explodedTbl) { _ =>
+              if (explodedTbl.getRowCount < 2) {
+                // we should never hit this
+                throw new GpuSplitAndRetryOOM(
+                  s"GPU OutOfMemory: cannot split a batch of 1 rows when " +
+                    s"the array has a single element!")
+              }
+              val splitIx = Seq(explodedTbl.getRowCount.toInt / 2)
+              explodedTbl.contiguousSplit(splitIx: _*)
+            }
+            val result = new Array[SpillableColumnarBatch](splits.size)
+            closeOnExcept(result) { _ =>
+              closeOnExcept(splits) { _ =>
+                (0 until splits.length).foreach { split =>
+                  val explodedTblToConvertBack = splits(split)
+                  val cols = new Array[ColumnVector](tbl.getNumberOfColumns)
+                  withResource(explodedTblToConvertBack) { _ =>
+                    val colToReconstitute = explodedTblToConvertBack.getTable.getColumn(0)
+                    closeOnExcept(cols) { _ =>
+                      (0 until tbl.getNumberOfColumns).foreach { col =>
+                        cols(col) = if (col == generatorOffset) {
+                          withResource(
+                            colToReconstitute.reduce(
+                              ReductionAggregation.collectList(), DType.LIST)) { sc =>
+                            ColumnVector.fromScalar(sc, 1)
+                          }
+                        } else {
+                          withResource(tbl.getColumn(col).getScalarElement(0)) { sc =>
+                            ColumnVector.fromScalar(sc, 1)
+                          }
```

Comment:
Seems like we're getting the first (and only) row of a column and then building a new column from that.  Why isn't this just incref of the original table column without copying anything?